hardware_interface/include/hardware_interface/hardware_component_interface.hpp

```diff
@@ -703,30 +703,117 @@ class HardwareComponentInterface : public rclcpp_lifecycle::node_interfaces::Lif
     return hardware_states_.find(interface_name) != hardware_states_.end();
   }
 
+  /// Get the state interface handle
+  /**
+   * \param[in] interface_name The name of the state interface to access.
+   * \return Shared pointer to the state interface handle.
+   * \throws std::runtime_error This method throws a runtime error if it cannot find the state
+   * interface with the given name.
+   */
+  const StateInterface::SharedPtr & get_state_interface_handle(
```

Comment:
Same as my prior comment if we now expose the StateIntefacePtr i think we should make the Handle handle the locking and not expose the lock

Reply:
We did it this way to have some flexibility from the user's point of view, and the main idea of this method is for the users to be able to catch the Interface in their own defined data structure and to avoid constant lookup in the unordered_map and sometimes this might be an expensive operation as showed in https://github.com/ros-controls/ros2_control/issues/2816

Reply:
i think this i now resolved